src/client/sandbox/code-instrumentation/properties/index.js

```diff
@@ -208,11 +209,20 @@ export default class PropertyAccessorsInstrumentation extends SandboxBase {
             },
 
             data: {
-                condition: el => domUtils.isDomElement(el) && domProcessor.isUrlAttr(el, 'data'),
+                // NOTE: The data property of the MessageEvent object cannot be redefined in the Android 5.1 browser
+                condition: el => domUtils.isDomElement(el) && domProcessor.isUrlAttr(el, 'data') ||
+                                 isAndroid && !nativeMethods.messageEventDataGetter && domUtils.isMessageEvent(el),
```

Comment:
Move `isAndroid && !nativeMethods.messageEventDataGetter && domUtils.isMessageEvent(el)` to the separate utils function `isSupportedMessageEvent`